From 4392ad501485965a4f3998f7841c0d1cf8959dea Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Mon, 11 Feb 2019 14:12:23 +0100
Subject: [PATCH] 71-nvme-iopolicy-netapp.rules: add default rules for NetApp
 products

NetApp ONTAP and NetApp E-Series controllers provide several paths to
the same subsystem, so we should be switching to 'round-robin' iopolicy
to provide the best performance.

Signed-off-by: Hannes Reinecke <hare@suse.com>
Signed-off-by: Simon Schricker <sschricker@suse.de>
---
 scripts/71-nvme-iopolicy-netapp-E-Series.rules | 2 ++
 scripts/71-nvme-iopolicy-netapp-ONTAP.rules    | 2 ++
 2 files changed, 4 insertions(+)
 create mode 100644 scripts/71-nvme-iopolicy-netapp-E-Series.rules
 create mode 100644 scripts/71-nvme-iopolicy-netapp-ONTAP.rules

diff --git a/scripts/71-nvme-iopolicy-netapp-E-Series.rules b/scripts/71-nvme-iopolicy-netapp-E-Series.rules
new file mode 100644
index 0000000..902c083
--- /dev/null
+++ b/scripts/71-nvme-iopolicy-netapp-E-Series.rules
@@ -0,0 +1,2 @@
+# Enable round-robin for NetApp E-Series
+ACTION=="add", SUBSYSTEM=="nvme-subsystem", ATTR{model}=="NetApp E-Series", ATTR{iopolicy}="round-robin"
diff --git a/scripts/71-nvme-iopolicy-netapp-ONTAP.rules b/scripts/71-nvme-iopolicy-netapp-ONTAP.rules
new file mode 100644
index 0000000..0d693a7
--- /dev/null
+++ b/scripts/71-nvme-iopolicy-netapp-ONTAP.rules
@@ -0,0 +1,2 @@
+# Enable round-robin for NetApp ONTAP
+ACTION=="add", SUBSYSTEM=="nvme-subsystem", ATTR{model}=="NetApp ONTAP Controller", ATTR{iopolicy}="round-robin"
-- 
2.22.0

